src/app/page.tsx

```diff
@@ -1,70 +1,75 @@
 "use client";
 import { useState, useEffect } from "react";
-import { budgetService } from "@/lib/budgetService";
 import Link from "next/link";
+import { monthService } from "@/lib/monthService";
+import { MonthlyBudget } from "@/types/budget";
 
-export default function HomePage() {
-  const [months, setMonths] = useState<string[]>([]);
+export default function Home() {
+  const [months, setMonths] = useState<MonthlyBudget[]>([]);
   const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
 
   useEffect(() => {
-    let mounted = true;
-    const fetchMonths = async () => {
+    async function fetchMonths() {
       try {
-        const allMonths = await budgetService.getAllMonths();
+        const allMonths = await monthService.getAllMonths();
         setMonths(allMonths);
       } catch (error) {
         console.error("Error loading months:", error);
+        setError("Failed to load months");
       } finally {
         setLoading(false);
       }
-    };
+    }
 
     fetchMonths();
-
-    return () => {
-      mounted = false;
-    };
   }, []);
 
-  if (loading) {
-    return <div className="p-6">Loading...</div>;
-  }
+  if (loading) return <div className="p-6">Loading...</div>;
+  if (error) return <div className="p-6 text-red-600">{error}</div>;
 
   return (
     <main className="p-6">
       <div className="flex justify-between items-center mb-6">
-        <h1 className="text-2xl font-bold">My Budget Tracker</h1>
+        <h1 className="text-2xl font-bold">Budget Months</h1>
         <Link
           href="/budget/new"
-          className="px-4 py-2 bg-red-600 text-white rounded hover:bg-red-700"
+          className="bg-blue-500 text-white px-4 py-2 rounded hover:bg-blue-600"
         >
-          Create New Month
+          New Month
         </Link>
       </div>
 
-      <div className="grid grid-cols-1 md:grid-cols-3 gap-4">
-        {months.length === 0 ? (
-          <p className="text-gray-500 col-span-3 text-center py-10">
-            No budget months created yet. Create your first month budget!
-          </p>
-        ) : (
-          months.map((month) => (
+      <div className="grid gap-4 md:grid-cols-2 lg:grid-cols-3">
+        {months.map((month) => {
+          const [year, monthNum] = month.month.split("-");
```

Comment:
instead of using *.split logic - you can normalize your data and keep year and month not in the string with separator, but in the object with few field. It will make you month-entity much more reusable and extensible